src/saveload/cheat_sl.cpp

```diff
@@ -14,37 +14,60 @@
 
 #include "../safeguards.h"
 
+static const SaveLoad _cheats_desc[] = {
+	SLE_VAR(Cheats, magic_bulldozer.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, magic_bulldozer.value, SLE_BOOL),
+	SLE_VAR(Cheats, switch_company.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, switch_company.value, SLE_BOOL),
+	SLE_VAR(Cheats, money.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, money.value, SLE_BOOL),
+	SLE_VAR(Cheats, crossing_tunnels.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, crossing_tunnels.value, SLE_BOOL),
+	SLE_NULL(1),
+	SLE_NULL(1),
+	SLE_VAR(Cheats, no_jetcrash.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, no_jetcrash.value, SLE_BOOL),
+	SLE_NULL(1),
+	SLE_NULL(1),
+	SLE_VAR(Cheats, change_date.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, change_date.value, SLE_BOOL),
+	SLE_VAR(Cheats, setup_prod.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, setup_prod.value, SLE_BOOL),
+	SLE_NULL(1),
+	SLE_NULL(1),
+	SLE_VAR(Cheats, edit_max_hl.been_used, SLE_BOOL),
+	SLE_VAR(Cheats, edit_max_hl.value, SLE_BOOL),
+};
+
 /**
  * Save the cheat values.
  */
 static void Save_CHTS()
 {
-	/* Cannot use lengthof because _cheats is of type Cheats, not Cheat */
-	byte count = sizeof(_cheats) / sizeof(Cheat);
-	Cheat *cht = (Cheat*) &_cheats;
-	Cheat *cht_last = &cht[count];
-
-	SlSetLength(count * 2);
-	for (; cht != cht_last; cht++) {
-		SlWriteByte(cht->been_used);
-		SlWriteByte(cht->value);
-	}
+	SlSetLength(std::size(_cheats_desc));
+	SlObject(&_cheats, _cheats_desc);
 }
 
 /**
  * Load the cheat values.
  */
 static void Load_CHTS()
 {
-	Cheat *cht = (Cheat*)&_cheats;
-	size_t count = SlGetFieldLength() / 2;
+	size_t count = SlGetFieldLength();
 	/* Cannot use lengthof because _cheats is of type Cheats, not Cheat */
-	if (count > sizeof(_cheats) / sizeof(Cheat)) SlErrorCorrupt("Too many cheat values");
+	if (count / 2 > sizeof(_cheats) / sizeof(Cheat)) SlErrorCorrupt("Too many cheat values");
```

Comment:
Maybe you could use `count > lengthof(_cheats_desc)`? Then you could also remove the nasty dummies.

Having said that, this commit does not seem to be part/mentioned in this PR. Are you trying to cheat it in? ;)
Or did you want to consider using SLE_STRUCT for each of the Cheats?

Reply:
This commit message making this change literal is called: `Codechange: rework CHTS-chunk save/load to be more like the others` :P I have to see if I have enough work to warrant another "mixed-fixes-for-stupid-code" PR .. but that is also why this PR is still a draft :)

Either way, this was existing code, I really did not parse yet what it said or tries to say. Just moved a line of code from one above to the one below.

What do you mean with "nasty dummies"? What "nasty" and what "dummies" is that referring to?

Reply:
For the dummies, take a look at: https://github.com/OpenTTD/OpenTTD/blob/1e564b333fa2d7bde2f21d7a05bb3e81305185c5/src/cheat_type.h#L31 further on line 33 and 36. That, together with the comment around line 23 made them nasty to me.